mapit/shortcuts.py

```diff
@@ -50,25 +50,24 @@ def output_json(out, code=200):
             out['debug_db_queries'] = connection.queries
         indent = 4
     encoder = GEOS_JSONEncoder(ensure_ascii=False, indent=indent)
-    content = encoder.iterencode(out)
 
-    # We don't want a generator function (iterencode) to be passed to an
-    # HttpResponse, as it won't cache due to its close() function adding it to
-    # an instance attribute.
-    content = map(lambda x: x, content)
+    if type(out) is dict:
+        response = http.JsonResponse(out, encoder=GEOS_JSONEncoder, status=code)
+    else:
+        content = encoder.iterencode(out)
 
-    types = {
-        400: http.HttpResponseBadRequest,
-        404: http.HttpResponseNotFound,
-        500: http.HttpResponseServerError,
-    }
-    response_type = types.get(code, http.StreamingHttpResponse)
+        # We don't want a generator function (iterencode) to be passed to an
+        # HttpResponse, as it won't cache due to its close() function adding it to
+        # an instance attribute.
+        content = map(lambda x: x, content)
+
+        response = http.StreamingHttpResponse(content_type='application/json; charset=utf-8', status=code)
```

Comment:
For consistency with Django's `JSONResponse`, do we want to drop the (superfluous) charset here to match?